engine/classes/ElggMenuItem.php

```diff
@@ -396,6 +399,39 @@ public function addLinkClass($class) {
 	}
 
 	/**
+	 * Set required AMD modules
+	 *
+	 * @param string[]|string $modules One or more required AMD modules
+	 * @return void
+	 */
+	public function setRequire($modules) {
+		$this->data['require'] = (array) $modules;
+	}
+
+	/**
+	 * Get required AMD modules
+	 *
+	 * @return string[]
+	 */
+	public function getRequire() {
+		$modules = (array) $this->data['require'];
+		return array_filter($modules, function($m) {
```

Comment:
Maybe even in __set, but need to check if value ends up in data...  or maybe not validate and log errors from elgg_require_js